src/backend/distributed/metadata/metadata_utility.c

```diff
@@ -1389,7 +1389,17 @@ ActiveShardPlacementList(uint64 shardId)
 	ShardPlacement *shardPlacement = NULL;
 	foreach_ptr(shardPlacement, shardPlacementList)
 	{
-		if (shardPlacement->shardState == SHARD_STATE_ACTIVE)
+		WorkerNode *workerNode =
+			FindWorkerNode(shardPlacement->nodeName, shardPlacement->nodePort);
```

Comment:
we should probably do some profiling at the end of the release cycle

Reply:
Yes, agreed. I'm also changing `SingleReplicatedTable` function [here](https://github.com/citusdata/citus/pull/5486/files#diff-565a339ed3c78bc5a0d4ffeb4e91032150b1dffbeeff59cd3e65981d20b998c7R473) as with replication > 1 and citus_disable_node we cannot only check the first shard and decide for hash distributed tables.